executor/src/witgen/generator.rs

```diff
@@ -31,12 +38,42 @@ pub struct Generator<'a, T: FieldElement> {
 impl<'a, T: FieldElement> Machine<'a, T> for Generator<'a, T> {
```

Comment:
Is `Generator` still the correct name for this thing?

Reply:
Probably not. It should also move into the `machines` module. But I'd suggest to do it in another PR, to keep the diff in bounds.